app/models/candidate_interface/personal_details_form.rb

```diff
@@ -70,15 +70,17 @@ def date_of_birth
 
       if valid_year?(year) && Date.valid_date?(*date_args)
         Date.new(*date_args)
+      else
+        OpenStruct.new(day: day, month: month, year: year)
       end
     end
 
     def date_of_birth_valid
-      errors.add(:date_of_birth, :invalid) if date_of_birth.nil?
+      errors.add(:date_of_birth, :invalid) if date_of_birth.class == OpenStruct
```

Comment:
My instinct is that it would be better to treat the OpenStruct here as a duck and ask the date whether any of its date, month or year fields are nil. Then we're not coupled to it

Reply:
Yes!

Reply:
I agree, however, a date can be invalid even if the fields are not nil. I changed this to `unless date_of_birth.is_a?(Date)` since we expect a valid date to be a Date, therefore coupling to that type does not feel harmful. Otherwise, I could create a helper method that can validate the date using `Date.valid_date?`. What do you think? @duncanjbrown

Reply:
:+1: I think this is good!